docs/asset-manager-framework.md

```diff
@@ -0,0 +1,316 @@
+    $registered_sizes['full'] = [
+        'width' => $image->width,
+        'height' => $image->height,
+        'crop' => false,
+    ];
+    if ( isset( $registered_sizes['medium'] ) ) {
+        $registered_sizes['medium']['crop'] = true;
+    }
+
+    $orientation = $image->height > $image->width ? 'portrait' : 'landscape';
+    $sizes = [];
+    foreach ( $registered_sizes as $name => $size ) {
+        // Unsplash uses imgix for resizing and modifying images.
+        $imgix_args = [
+            'w' => $size['width'],
+            'h' => $size['height'],
+            'fit' => $size['crop'] ? 'crop' : 'max',
+        ];
+        $sizes[ $name ] = [
+            'width' => $size['width'],
+            'height' => $size['height'],
+            'orientation' => $orientation,
+            'url' => add_query_arg( urlencode_deep( $imgix_args ), $image->urls->raw ),
+        ];
+    }
+
+    $item->set_sizes( $sizes );
+
+    return $item;
+}
+```
+
+**`set_filename( string $filename )`**
+
+Sets the media item's filename.
+
+**`set_image( string $image )`**
+
+Sets a placeholder image or custom thumbnail URL, useful for non-image based media.
+
+**`set_link( string $link )`**
+
+Sets a link to a URL to view the media if available. This is equivalent to the attachment page view in WordPress.
+
+**`set_alt( string $alt )`**
+
+Sets the alternative text for the media, commonly used in the `alt` attribute.
+
+**`set_description( string $description )`**
+
+Sets the description text for the media, can be used for the caption in some cases but is intended for a long description of the file.
+
+**`set_caption( string $caption )`**
+
+Sets caption text for the media, commonly displayed below the media in post content.
+
+**`set_name( string $name )`**
+
+Sets the slug or URL safe version of the media title for use in links.
+
+**`set_date( int $date )`**
+
+A unix timestamp of when the file was created.
+
+**`set_modified( int $modified )`**
+
+A unix timestamp of when the file was last edited.
+
+**`set_file_size( int $file_size )`**
+
+Sets the file size in bytes.
+
+**`set_author( string $name, string $link )`**
+
+Set the media file's author name and an optional link to their website or other appropriate URL.
+
+**`set_meta( array $meta )`**
+
+Set a key/value list of meta data associated with the media, stored as post meta when using the media.
+
+**`add_meta( string $key, mixed $value )`**
+
+Adds an item of meta data rather than setting or overwriting data like `set_meta()`.
+
+**`set_amf_meta( array $meta )`**
+
+Set additional meta data not stored in the database but for use within AMF during processing. See the hooks and filters section below.
+
+**`add_amf_meta( string $key, mixed $value )`**
+
+Adds an item of AMF only meta data rather than setting or overwriting data like `set_amf_meta()`.
+
+### `AssetManagerFramework\Image` Class
+
+Extends the `AssetManagerFramework\Media` class.
```

Comment:
I think ultimately more info could be given on these class descriptions (e.g. for intended use, more than just "this extends that other thing"), but probably not essential right now

Reply:
Maybe, the classes themselves documented like this are basically just aliases for `Media` right now so there isn't any real difference.